integrated_channels/integrated_channel/learner_data.py

```diff
@@ -134,6 +133,52 @@ def collect_learner_data(self):
                 is_passing=is_passing,
             )
 
+    def _should_skip_enrollments_data_collection(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection of the enterprise enrollment due to any configuration reason.
+
+        Reasons for skipping data collection may include data sharing consent not being granted,
+        or audit track data reporting being disabled for audit enrollments, etc..
+
+        :param enterprise_enrollment: The enterprise enrollment whose configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        return any([
+            not enterprise_enrollment.consent_available,
+            self._disabled_audit_reporting(enterprise_enrollment),
+        ])
+
+    def _disabled_audit_reporting(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection due to audit track data reporting not being enabled.
+
+        If it is enabled, we simply skip the check, regardless of the ``enterprise_enrollment``.
+        If it isn't enabled, we check to see if the course mode is of an audit type.
+
+        :param enterprise_enrollment: The enterprise enrollment whose reporting configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
```

Comment:
@UmanShahzad A couple suggestions for this docstring:

```
Specifies whether audit track data reporting is disabled for `enterprise_enrollment`.

* If the current Enterprise customer enables audit track data reporting, simply return False.
* If the current Enterprise customer does not enable audit track data reporting,
  return True if we are dealing with an audit enrollment, and False otherwise.

:param: enterprise_enrollment: The enterprise enrollment whose reporting configuration is to be checked against.
:return: True if audit track data reporting is disabled, False otherwise.